### DISCUSSION TITLE
NvPCF seems not handled properly by the driver

### DISCUSSION THREAD

**Discussion by 500InternalError** [NVIDIA/open-gpu-kernel-modules#677] [Q&A] [answered]
On some newer systems like asus GA403 series, it seems that NPCF object have event `0xC1`:
```
                If ((NTWF == One))
                {
                    Notify (NPCF, 0xC1) // Hardware-Specific
                    Acquire (MUT0, 0x1000)
                    NTWF = Zero
                    Release (MUT0)
```

However, according to `kernel-open/nvidia/nv-acpi.c`:
```
static void nv_acpi_nvpcf_event(acpi_handle handle, u32 event_type, void *data)
{
    nv_acpi_t *pNvAcpiObject = data;

    if (event_type == ACPI_NVPCF_EVENT_CHANGE) // ACPI_NVPCF_EVENT_CHANGE==0xC0
    {
        rm_acpi_nvpcf_notify(pNvAcpiObject->sp);
    }
    else
    {
        nv_printf(NV_DBG_INFO,"NVRM: %s: NVPCF event 0x%x is not supported\n", event_type, __FUNCTION__);
    }
}
```
it seems that the driver does not handle event `0xC1` at all, which causes nvidia-powerd won't work on these systems. Will nvidia fix this?

**Comment by aritger** [comment]
Thank you for your report.  Yes, we're aware of the 0xC1 event but we don't currently support it in the Linux driver.

To double check that this is the same thing we're already aware of, could you please generate and attach:
- nvidia-bug-report.log.gz
- the output of `acpidump`

Thanks.

**Comment by 500InternalError** [comment]
[nvidia-bug-report.log.gz](https://github.com/user-attachments/files/16276111/nvidia-bug-report.log.gz)
[acpidump.txt](https://github.com/user-attachments/files/16276113/acpidump.txt)

Do you mind tell me when will nvidia fix this? It's really a pain in ass for users like me, I only get 55W rather than 90W for GPU.

**Comment by aritger** [comment, marked as answer]
Thanks for the logs.  I don't currently have an ETA for a fix, but the feedback here helps our prioritization, so thank you.

**Reply by flukejones** [reply]
Just a heads up: I have a discord server catering to ASUS gaming laptops on Linux (I do the bulk of support work in kernel) and I see more than a few (guesstimate 200+) users with 2024 models of the G14 and G16 who are very severely handicapped by low power limits. These are the GA403, GU605, and every likely the GA605 when it releases.

I see messages about this issue at least every few days, and never from the same people - all of whom are expressing a great deal of frustration.

**Reply by aritger** [reply]
I /think/ this should be fixed in the new 560.35.03 NVIDIA GPU driver released today.  Can you please test to verify if the fix is working in your case?

**Comment by vindex10** [comment]
I'm sorry if it turns out to be not related, but I also have Asus gaming laptop and nvidia driver causes stattering when the GPU is enabled. `nv_acpi_nvpcf_event` appears in the kernel stack trace, so I assumed results of my investigation could be useful here :)

https://forums.developer.nvidia.com/t/random-freezes-rtx-4070-due-to-kworker-acpi-interrupts-in-ubuntu-6-8/293192

**Reply by aritger** [reply]
That symptom sounds different than the issue described by the other posters in this thread.  I see @amrit1711 is communicating with you in the developer.nvidia.com forum, so I'll leave that discussion there.
